lib/galaxy/datatypes/binary.py

```diff
@@ -4864,3 +4847,105 @@ def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> N
         with open(dataset.get_file_name(), "rb") as handle:
             header_bytes = handle.read(8)
         dataset.metadata.version = struct.unpack("<i", header_bytes[4:8])[0]
+
+
+@build_sniff_from_prefix
+class Safetensors(Binary):
+    """
+    safetensors is a new simple format for storing tensors safely (as opposed to pickle) and that is still fast (zero-copy).
+    It provides a secure way to store and load tensors without the security risks associated with pickle-based formats.
+    Safetensors files consist of a JSON header followed by tensor data.
+    more info at: https://github.com/huggingface/safetensors
+    """
+
+    file_ext = "safetensors"
+
+    def __init__(self, **kwd):
+        super().__init__(**kwd)
+
+    def sniff_prefix(self, file_prefix: FilePrefix) -> bool:
+        """
+        Determining if the file is in safetensors format
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('cellpose_model_safetensors.safetensors')
+        >>> Safetensors().sniff(fname)
+        True
+        >>> fname = get_test_fname('cellpose_model_pytorch_pickle.pth')
```

Comment:
I don't think the pth file is similar enough to safetensor, I'd just use an already included binary file:

```suggestion
        >>> fname = get_test_fname("lib/galaxy/datatypes/test/test_charmm.vel")
        >>> Safetensors().sniff(fname)
        False
```

and remove `lib/galaxy/datatypes/test/cellpose_model_pytorch_pickle.pth`